libwayshot/src/lib.rs

```diff
@@ -413,7 +413,13 @@ impl WayshotConnection {
         Ok(frame_copies)
     }
 
-    fn overlay_frames(&self, frames: &[(FrameCopy, FrameGuard, OutputInfo)]) -> Result<()> {
+    /// Create a layer shell surface for each output,
+    /// render the screen captures on them and use the callback to select a region from them
+    fn overlay_frames_and_select_region(
+        &self,
+        frames: &[(FrameCopy, FrameGuard, OutputInfo)],
+        callback: Box<dyn Fn() -> Result<LogicalRegion, Error>>,
+    ) -> Result<(LogicalRegion)> {
```

Comment:
```suggestion
    ) -> Result<LogicalRegion> {
```

@Shinyzenith it seems like we don't have clippy enabled in GitHub actions?

```
   Compiling wayshot v1.3.2-dev (/home/andreas/dev/wayshot/wayshot)
warning: unused import: `Region`
  --> libwayshot/src/lib.rs:55:29
   |
55 |     region::{LogicalRegion, Region, Size},
   |                             ^^^^^^
   |
   = note: `#[warn(unused_imports)]` on by default

warning: unnecessary parentheses around type
   --> libwayshot/src/lib.rs:422:17
    |
422 |     ) -> Result<(LogicalRegion)> {
    |                 ^             ^
    |
    = note: `#[warn(unused_parens)]` on by default
help: remove these parentheses
    |
422 -     ) -> Result<(LogicalRegion)> {
422 +     ) -> Result<LogicalRegion> {
    |

warning: unused import: `GenericImageView`
 --> libwayshot/src/image_util.rs:1:27
  |
1 | use image::{DynamicImage, GenericImageView};
  |                           ^^^^^^^^^^^^^^^^

warning: unused variable: `logical_height`
  --> libwayshot/src/image_util.rs:15:25
   |
15 |     let (logical_width, logical_height) = match transform {
   |                         ^^^^^^^^^^^^^^ help: if this is intentional, prefix it with an underscore: `_logical_height`
   |
   = note: `#[warn(unused_variables)]` on by default

warning: `libwayshot` (lib) generated 4 warnings (run `cargo fix --lib -p libwayshot` to apply 3 suggestions)
warning: unused import: `fs::File`
 --> wayshot/src/wayshot.rs:2:5
  |
2 |     fs::File,
  |     ^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default

warning: `wayshot` (bin "wayshot") generated 1 warning (run `cargo fix --bin "wayshot"` to apply 1 suggestion)
    Finished release [optimized] target(s) in 0.92s
     Running `target/release/wayshot --clipboard --slurp ''`
Error: error occurred in freeze callback

Location:
    wayshot/src/wayshot.rs:72:9
```

Shouldn't impact the PR, but wanted to mention.

Reply:
Will address this.